File: language-examples/csharp/rest/RestDeleteDocument.cs
using System.Text.Json;

class RestDeleteDocuement
{
    /// <summary>
    /// Deletes a document from a corpus using delete-doc API.
    /// </summary>
    /// <param name="customerId"> The unique customer ID in Vectara platform. </param>
    /// <param name="corpusId"> The corpus ID to which data will be indexed. </param>
    /// <param name="jwtToken"> A valid authentication token. </param>
    /// <param name="docId"> Id of the document that needs to be deleted.
    public static void DeleteDocument(long customerId, 
                                      long corpusId, 
                                      string jwtToken, 
                                      string docId)
    {
        using (var client = new HttpClient())
        {
            try
            {
                var request = new HttpRequestMessage
                {
                    RequestUri = new Uri($"https://{ServerEndpoints.commonEndpoint}/v1/delete-doc"),
                    Method = HttpMethod.Post,
                };

                Dictionary<string, object> deleteRequest = new()
                {
                    { "customerId", customerId },
                    { "corpusId", corpusId },
                    { "documentId", docId }
                };
                string jsonData = JsonSerializer.Serialize(deleteRequest);

                request.Content = new StringContent(jsonData);
                request.Content.Headers.Remove("Content-Type");
                request.Content.Headers.Add("Content-Type", "application/json");

                request.Headers.Add("customer-id", customerId.ToString());
                request.Headers.Add("Authorization", $"Bearer {jwtToken}");

                HttpResponseMessage response = client.Send(request);
                string result = response.Content.ReadAsStringAsync().Result;

                Console.WriteLine(result);
            }
            catch (Exception)
            {
                throw;
            }
        }
    }
}

File: language-examples/python/vectara-rest/rest_delete_document.py
"""Simple example of using the Vectara REST API for deleting a document."""

import json
import logging
import requests

def _get_delete_request_json(customer_id: int, corpus_id: int, doc_id: str):
    """Returns json delete request."""
    request = {}
    request['customer_id'] = customer_id
    request['corpus_id'] = corpus_id
    request['document_id'] = doc_id

    return json.dumps(request)

def delete_document(customer_id: int, corpus_id: int, idx_address: str, jwt_token: str, doc_id: str):
    """Deletes document from the corpus.
    Args:
        customer_id: Unique customer ID in vectara platform.
        corpus_id: ID of the corpus from which document willb e deleted.
        idx_address: Address of the indexing server. e.g., api.vectara.io
        jwt_token: A valid Auth token.
        doc_id: Id of the document to be deleted.

    Returns:
        (response, True) in case of success and returns (response, False) in case of failure.
    """

    post_headers = {
        "Authorization": f"Bearer {jwt_token}",
        "customer-id": f"{customer_id}"
    }
    response = requests.post(
        f"https://{idx_address}/v1/delete-doc",
        data=_get_delete_request_json(customer_id, corpus_id, doc_id),
        verify=True,
        headers=post_headers)

    if response.status_code != 200:
        logging.error("REST delete document failed with code %d, reason %s, text %s",
                       response.status_code,
                       response.reason,
                       response.text)
        return response, False
    return response, True